server/index.ts

```diff
@@ -19,22 +37,48 @@ app.use(express.json());
 
 const port = process.env.PORT || 8080;
 
-app.get("/product_data", (_, res: Response) => {
+// TODO: Implement this
+const authMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const cookies = parseCookies(req);
+  const sessionToken = cookies[DescopeClient.SessionTokenCookieName];
+
+  try {
+    await clientAuth.validateSession(sessionToken);
```

Comment:
I had to wrap this with setting a variable to work for me. Might have been something else though.

Reply:
Yeah, I'm not sure why, I mean I'm not doing it and it worked for me. It shouldn't need that right, since as long as the Promise is returned as fulfilled, it should exit the try/catch right?